src/workflow_components/composite_action.py

```diff
@@ -131,29 +160,43 @@ class CompositeAction(GraphObject):
     image = Property()
     url = Property()
     is_public = Property()
+    refs = Property()
+    commit_sha = Property()
 
     composite_action_input = RelatedTo(CompositeActionInput)
     steps = RelatedTo(CompositeActionStep)
 
-    def __init__(self, name: Optional[str], path: str):
+    def __init__(self, name: Optional[str], path: str, commit_sha: str):
         self.name = name
         self.path = path
-        self._id = md5(path.encode()).hexdigest()
+        self.commit_sha = commit_sha
+        self._id = md5(f"{path}_{commit_sha}".encode()).hexdigest()
+        self.refs = []
 
     @staticmethod
     def from_dict(obj_dict) -> "CompositeAction":
-        ca = CompositeAction(name=obj_dict.get("name"), path=obj_dict["path"])
+        ca = CompositeAction(
+            name=obj_dict.get("name"),
+            path=obj_dict["path"],
+            commit_sha=obj_dict["commit_sha"],
+        )
 
         ca.url = obj_dict["url"]
         ca.is_public = obj_dict["is_public"]
         if "inputs" in obj_dict:
             for name, input in obj_dict["inputs"].items():
-                input["_id"] = md5(f"{ca._id}_{name}".encode()).hexdigest()
+                input["_id"] = md5(
+                    f"{ca._id}_{name}_{ca.commit_sha}".encode()
```

Comment:
Please make sure you changed all test files that calculate the node ID in the old form